### PR TITLE
UI tests to swipe through every category and source

### PR DESCRIPTION
**Pull request by alexeystrakh**
This PR includes two new UI tests:

1. swipe through each category, ensure the title and sequence
2. swipe through each source, ensure the title and sequence

swipe_categories

swipe_sources

### CODE REVIEW

File: samples/news/News.UITests/News.UITests.csproj [hunk 1]
```diff
@@ -49,6 +49,10 @@
       <ReferenceOutputAssembly>False</ReferenceOutputAssembly>
       <Private>False</Private>
     </ProjectReference>
+    <ProjectReference Include="..\News\News.csproj">
```
Comment (alexeystrakh) [inline samples/news/News.UITests/News.UITests.csproj]: this reference is a new one to access common constants (list of known categories and sources) located in the shared core project

File: samples/news/News.UITests/Pages/SourcesPage.cs [hunk 2]
```diff
@@ -9,8 +15,27 @@ public class SourcesPage : BasePage
...
+        public List<string> SupportedSources => Constants.Sources.All;
+
         public SourcesPage()
         {
+            SourceLabel = x => x.Marked(nameof(SourceLabel));
+        }
+
+        public void ShowNextSource()
+        {
+            app.SwipeRightToLeft(swipeSpeed: 1000);
```
Comment (alexeystrakh) [inline samples/news/News.UITests/Pages/SourcesPage.cs]: with this speed, it looks more natural than with the default 500

File: samples/news/News.UITests/Tests.cs [hunk 3]
```diff
@@ -62,6 +62,31 @@ public void SetLightMode()
                 .SetLightMode();
         }
 
+        [Test]
+        public void SwipeThroughEachCategory()
+        {
+            var homePage = new HomePage();
```
Comment (Sweekriti91) [inline samples/news/News.UITests/Tests.cs]: I don't really like this syntax and style of test writing, it makes no sense to me what is happening. I want the tests to be following the pattern we have and the idea being they are easily readable and give a sense of what is being tested. This is the syntax pattern our tests should follow : 

```
 new HomePage()
         .ShowSearchTab()
         .Search("Microsoft");
```
Also loops inside a test is asking for trouble ,if one category fails, the rest of the test will fail. Small and independent tests is what I think is the best approach. 

My suggestion would be to use NUnit [Category] tags and remove the foreach loop. Divide this into separate tests. 

Why independent tests are important (Also referenced in my blog post) :
https://github.com/xamarin-automation-service/uitest-pop-example/wiki/General-Testing-Suggestions#split-tests-up-into-small-chunks-wherever-possible 

NUnit Category Tag : http://nunit.org/docs/2.2/category.html

File: samples/news/News.UITests/Tests.cs [hunk 4]
```diff
@@ -62,6 +62,31 @@ public void SetLightMode()
...
+            foreach (var selectedCategory in newsPage.SupportedCategories)
+            {
+                newsPage.ValidateCategory(selectedCategory);
+                newsPage.ShowNextCategory();
+            }
+        }
+
+        [Test]
+        public void SwipeThroughEachSource()
+        {
```
Comment (Sweekriti91) [inline samples/news/News.UITests/Tests.cs]: Same comments from `SwipeThroughEachCategory`

**Comment by Sweekriti91** [comment]
@DeanFaizal is this your test? 

```
public void ViewEachTabInDarkModeAndLightMode()
        {
            var homePage = new HomePage();
            homePage.ShowNewsTab();
            homePage.ShowSourcesTab();
            homePage.ShowFavoritesTab();
            homePage.ShowSearchTab();
            homePage.ShowSettingsTab()
            .SetLightMode();

            homePage.ShowNewsTab();
            homePage.ShowSourcesTab();
            homePage.ShowFavoritesTab();
            homePage.ShowSearchTab();
        }
```

This is also incorrect to the POP style of writing tests, could we update this to match the correct pattern?

File: samples/news/News.UITests/Tests.cs (hunk 3 above)
Comment (alexeystrakh) [reply samples/news/News.UITests/Tests.cs]: that is exactly the pattern we used to have so I was following the existing test:

commit-uitests

https://github.com/xamarin/mobcat/commit/d5e273d61f6471c6eb8ff74e2b99c6c91923b178#diff-2e95bafb133ac0253594a183cb8dae8a

Comment (alexeystrakh) [reply samples/news/News.UITests/Tests.cs]: I've reimplemented test and created a new instance of a page object whenever it's out of execution chain. Please check the updated source

Comment (alexeystrakh) [reply samples/news/News.UITests/Tests.cs]: > Also loops inside a test is asking for trouble, if one category fails, the rest of the test will fail. Small and independent tests are what I think is the best approach.

this test is testing not a swipe action but sequence of categories and sources. Basically, this is the smallest chunk possible to test that all categories and sources are present in a given number and order. Is it possible to implement it with the `NUnit Category Tag`?

File: samples/news/News.UITests/Pages/SourcesPage.cs [hunk 5]
```diff
@@ -9,8 +15,28 @@ public class SourcesPage : BasePage
...
+        protected readonly Query SourceLabel;
+
+        public List<string> SupportedSources => Constants.Sources.All;
+
         public SourcesPage()
         {
+            SourceLabel = x => x.Marked(nameof(SourceLabel));
+        }
+
+        public void ShowNextSource()
```
Comment (Sweekriti91) [inline samples/news/News.UITests/Pages/SourcesPage.cs]: why is the return type `void`? we aren't leaving the SourcePage (Tab) right? we stay in the same tab? this should return SourcePage. 

`void` is for navigation away from the current page, or in the case of this app, navigation to a different tab

Comment (alexeystrakh) [reply samples/news/News.UITests/Pages/SourcesPage.cs]: It was done by following the example in the `AddFavorite` test (which was already implemented earlier and served as an architectural decision for me):

```
   public void AddFavorite()
        {
            app.WaitForElement(FavoriteButton);
            app.Tap(FavoriteButton);
            app.Screenshot("Added favorite article");
        }
```

Here we being at the News page adding to favorites, no navigation occurs, the return type is this void.

Comment (alexeystrakh) [reply samples/news/News.UITests/Pages/SourcesPage.cs]: Let me update all the test to reflect the `navigation away` logic.

**Pull request merged**